lamp.py

```diff
@@ -0,0 +1,209 @@
+
+    toolchain_dir = os.path.join(install_dir, 'gennytoolchain')
+    if os.path.exists(toolchain_dir):
+        logging.info('Skipping extracting the toolchain into: %s', toolchain_dir)
+    else:
+        logging.info('Extracting gennytoolchain, please wait...')
+        tarball = tarfile.open(toolchain_tarball)
+        tarball.extractall(toolchain_dir)
+        logging.info('Finished extracting gennytoolchain into %s', toolchain_dir)
+
+    return toolchain_dir
+
+
+def _create_compile_environment(toolchain_dir, triplet_os):
+    env = os.environ.copy()
+    paths = [env['PATH']]
+
+    # For mongodbtoolchain compiler (if there).
+    paths.insert(0, '/opt/mongodbtoolchain/v3/bin')
+
+    # For cmake and ctest
+    cmake_bin_relative_dir = {
+        'linux': 'downloads/tools/cmake-3.13.3-linux/cmake-3.13.3-Linux-x86_64/bin',
+        'osx': 'downloads/tools/cmake-3.13.3-osx/cmake-3.13.3-Darwin-x86_64/CMake.app/Contents/bin'
+    }[triplet_os]
+    paths.insert(0, os.path.join(toolchain_dir, cmake_bin_relative_dir))
+
+    # For ninja
+    ninja_bin_dir = os.path.join(toolchain_dir,
+                                 'downloads/tools/ninja-1.8.2-{}:'.format(triplet_os))
+    paths.insert(0, ninja_bin_dir)
+
+    env['PATH'] = ':'.join(paths)
+    env['NINJA_STATUS'] = '[%f/%t (%p) %es]'  # make the ninja output even nicer
+
+    logging.debug('Using environment: %s', env)
+    return env
+
+
+def cmake(toolchain_dir, cmdline_cmake_args, triplet_os, env):
+    cmake_cmd = ['cmake', '-B', 'build', '-G', 'Ninja']
+    # We set both the prefix path and the toolchain file here as a hack to allow cmake
+    # to find both shared and static libraries. vcpkg doesn't natively support a project
+    # using both.
+    cmake_prefix_path = os.path.join(toolchain_dir, 'installed/x64-{}-shared'.format(triplet_os))
+    cmake_toolchain_file = os.path.join(toolchain_dir, 'scripts/buildsystems/vcpkg.cmake')
+
+    cmake_cmd += [
+        '-DCMAKE_PREFIX_PATH={}'.format(cmake_prefix_path),
+        '-DCMAKE_TOOLCHAIN_FILE={}'.format(cmake_toolchain_file),
+        '-DVCPKG_TARGET_TRIPLET=x64-{}-static'.format(triplet_os),
+    ]
+
+    cmake_cmd += cmdline_cmake_args
+
+    logging.info('Running cmake: %s', cmake_cmd)
+    subprocess.run(cmake_cmd, env=env)
+
+
+def compile(env):
+    compile_cmd = ['ninja', '-C', 'build']
+    logging.info('Running ninja: %s', compile_cmd)
+    subprocess.run(compile_cmd, env=env)
+
+
+def install(env):
+    install_cmd = ['ninja', '-C', 'build', 'install']
+    logging.info('Running ninja install: %s', install_cmd)
+    subprocess.run(install_cmd, env=env)
+
+
+def cmake_test(env):
+    workdir = os.path.join(os.getcwd(), 'build')
+    ctest_cmd = [
+        'ctest',
+        '--label-exclude',
+        '(standalone|sharded|single_node_replset|three_node_replset)'
+    ]
+    subprocess.run(ctest_cmd, cwd=workdir, env=env)
+
+
+def parse_args(args, os_family):
+    parser = argparse.ArgumentParser(
```

Comment:
It's worth moving the parser into the module namespace, it's a nice step toward composable argparse parsers.